### ISSUE TITLE
Questions about masking tokens of labels

### ISSUE THREAD

**Issue by noobpeng99** [FactoDeepLearning/DAN#18]
Hello！

Thank you for your sharing.

I noticed that the code you provided uses a batch size of 1 to train the model. When I use a batch size of 2 or more, I need to pad the tokens in Groundtruth. Usually, this means that the padding needs to be masked in the Decoder. This seems to be achieved via the `generate_token_mask` at line 346~354 in  `models_dan.py`. 

```python
    def generate_token_mask(self, token_len, total_size, device):
        """
        Generate mask for tokens per sample
        """
        batch_size, len_max = total_size
        mask = torch.zeros((batch_size, len_max), dtype=torch.bool, device=device)
        for i, len_ in enumerate(token_len):
            mask[i, :len_] = False
        return mask
```
However, this method seems to return a matrix of shape (batch_size, len_max) whose elements are all False. This results in the following not working according to this mask.

I think it should be changed to `mask = torch.ones((batch_size, len_max), dtype=torch.bool, device=device)` . 

Could you please provide more explanation, I would be grateful

**Comment by FactoDeepLearning** [comment]
You must be right, indeed.

As you mentioned, the code was only tested for a batch size of 1... You also need to check this line of code:
https://github.com/FactoDeepLearning/DAN/blob/da3046a1cc83e9be3e54dd31a5e74d6134d1ebdc/OCR/document_OCR/dan/trainer_dan.py#L71C3-L71C3

**Comment by noobpeng99** [comment]
> You must be right, indeed.
> 
> As you mentioned, the code was only tested for a batch size of 1... You also need to check this line of code: https://github.com/FactoDeepLearning/DAN/blob/da3046a1cc83e9be3e54dd31a5e74d6134d1ebdc/OCR/document_OCR/dan/trainer_dan.py#L71C3-L71C3

Thank you for your kind reply, it is very helpful to me
